firedrake/preconditioners/patch.py

```diff
@@ -436,12 +435,13 @@ def make_residual_wrapper(coeffs, maps):
    if (x) {{
      ierr = VecRestoreArrayRead(x, &state);CHKERRQ(ierr);
    }}
+   PetscLogFlops({} * npoints);
```

Comment:
Would it not be better to just omit this line if `flops` is not provided? I think defaulting to zero is potentially a bit confusing.

Reply:
Let's not make it optional then.